test/TrialsTests.jl

```diff
@@ -233,17 +237,92 @@ else
+s001 = sprint(show, "text/plain", t001)
+@test contains(s001, "┌ Trial:")  # box starting at the type
+@test contains(s001, "│  time 3.142 ms")
+@test contains(s001, "│  0 allocations\n")  # doesn't print 0 bytes after this
+@test contains(s001, "└  1 sample, with 1 evaluation")
+
+# Histogram utils
+@test BenchmarkTools.asciihist([1,2,3]) == ['▃' '▆' '█']
+@test BenchmarkTools.asciihist([1,2,0,3], 2) == [' ' '▃' ' ' '█'; '▇' '█' '▁' '█']
+
+@test BenchmarkTools.histogram_bindata([1.1, 3.1, 99], 1:3) == [1,0,2]
+@test BenchmarkTools.histogram_bindata([1.1, -99, 3.1], 1:3.0) == [1,0,1]
+
+# Trials with several samples
+t003 = BenchmarkTools.Trial(BenchmarkTools.Parameters(), [0.01, 0.02, 0.04], [0,0,0], 0, 0)
+s003 = sprint(show, "text/plain", t003)
+@test contains(s003, " 1 ns +")  # right limit is 1ns
+@test contains(s003, "min 0.010 ns, median 0.020 ns, mean 0.023 ns, 99ᵗʰ 0.040 ns")
+
+t123 = BenchmarkTools.Trial(BenchmarkTools.Parameters(), [1,2,3.], [0,0,0.], 0, 0)
+s123 = sprint(show, "text/plain", t123)
+@test contains(s123, "min 1.000 ns, median 2.000 ns, mean 2.000 ns")
+@test contains(s123, " 0 allocations\n")  # doesn't print 0 bytes after this
+@test contains(s123, " ◑* ")  # median ◑ is shifted left
+@test contains(s123, "▁▁█▁▁")  # has a histogram, mostly zero
+@test contains(s123, "▁▁▁█ ▁\n")  # 3.0 fits in last bin, not the overflow
+@test endswith(s123, "3 ns +")  # right endpoint rounded to 3, no decimals
+@test contains(s123, "3 samples, each 1 evaluation")  # caption
+
+t456 = BenchmarkTools.Trial(BenchmarkTools.Parameters(), 100 * [1,1,3,14,16.], [0,0,2,0,0.], 456, 7)
+s456 = sprint(show, "text/plain", t456)
+@test contains(s456, "7 allocations, total 456 bytes")
+@test contains(s456, "GC time: mean 0.400 ns (0.06%), max 2.000 ns (0.67%)")
+@test contains(s456, "┌ Trial:")  # box starting at the type
+@test contains(s456, "│  ◔       ")  # 1st quartile lines up with bar
+@test contains(s456, "│  █▁▁▁▁▁▁▁")
+@test contains(s456, "└  100 ns ")  # box closing + left endpoint without decimals
+
+# Compact show & arrays of Trials
+@test sprint(show, t001) == "Trial(3.142 ms)"
+@test sprint(show, t003) == "Trial(0.010 ns)"
+@test sprint(show, "text/plain", [t001, t003]) == "2-element Vector{BenchmarkTools.Trial}:\n 3.142 ms\n 0.010 ns"
+@test_skip sprint(show, "text/plain", [t0]) == "1-element Vector{BenchmarkTools.Trial}:\n ??"
+
+#=
+
+# Some visual histogram checks, in which mean/median should highlight a bar, or not:
+
+using BenchmarkTools: Trial, Parameters
+Trial(Parameters(), [pi * 10^9], [0], 0, 0)  # one sample
+
+# mean == median, one bar. Symbol for median moves to the left.
+Trial(Parameters(), [pi, pi], [0, 0], 0, 0)
+Trial(Parameters(), fill(101, 33), vcat(zeros(32), 50), 0, 0)
+
+# mean == median, three bars
+Trial(Parameters(), [3,4,5], [0,0,0], 0, 0)
+
+# three bars, including mean not median
+Trial(Parameters(), pi * [1,3,4,4], [0,0,0,100], 1, 1)
```

Comment:
These examples previously highlighted the wrong bar as median. This was computed by hand based on the sorted times, which I think worked fine in most cases. Now it uses `searchsortedlast(fences, t)` with `t = median(times)`, the exact same method as used for filling in the bars themselves. 

Could try a bit harder to write automatic tests for these, maybe. I believe tests above should catch obvious things, like changes which introduce off-by-1 errors in marker positions.